core/src/main/java/org/elasticsearch/rest/action/support/RestStatusToXContentListener.java

```diff
@@ -23,14 +23,30 @@
 import org.elasticsearch.rest.BytesRestResponse;
 import org.elasticsearch.rest.RestChannel;
 import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.RestStatus;
+
+import java.util.function.Function;
 
 /**
- *
+ * Content listener that extracts that {@link RestStatus} from the response.
  */
 public class RestStatusToXContentListener<Response extends StatusToXContent> extends RestResponseListener<Response> {
+    private final Function<Response, String> extractLocation;
 
+    /**
+     * Build an instance that doesn't support responses with the status {@code 201 CREATED}.
+     */
     public RestStatusToXContentListener(RestChannel channel) {
+        // TODO switch this to throwing an exception
+        this(channel, r -> "unsupported");
```

Comment:
wouldn't this cause adding "Location" : "unsupported" header whenever 201 is returned apart from index and update api? If so, although this is temporary, maybe we could prevent that from happening?